selectors/lhs.ts

```diff
@@ -1,8 +1,64 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
+import {createSelector} from 'reselect';
 import {GlobalState} from 'types/store';
+import {StaticPage} from 'types/store/lhs';
+import {makeGetDraftsCount} from 'selectors/drafts';
+import {
+    insightsAreEnabled,
+    isCollapsedThreadsEnabled,
+    localDraftsAreEnabled,
+} from 'mattermost-redux/selectors/entities/preferences';
 
 export function getIsLhsOpen(state: GlobalState): boolean {
     return state.views.lhs.isOpen;
 }
+
+export function getCurrentStaticPageId(state: GlobalState): string {
+    return state.views.lhs.currentStaticPageId;
+}
+
+export const getDraftsCount = makeGetDraftsCount();
+
+export const getVisibleStaticPages = createSelector(
+    'getVisibleSidebarStaticPages',
+    (state: GlobalState) => {
```

Comment:
This needs to be done in the result function (the last argument to `createSelector`). As-written, this selector is called every time `getVisibleStaticPages` is, and because it returns a new array, the selector recalculates the result function every time this is called.

That's easy enough to fix though by just passing `insightsAreEnabled` and such separately and then just building `pages` in that final function.

Reply:
Oh valid point, and also this way we don't need to loop to find drafts to change the visibility based on the count. Thanks!